### PR TITLE
wluma: 4.1.0 -> 4.1.2

### PR DESCRIPTION
**Pull request by yshym**
###### Motivation for this change

Update to the latest version: https://github.com/maximbaz/wluma/releases/tag/4.1.2
It seems like we don't need to add any build parameters, because we have `vulkan-loader` vulkan dependency only, which is required (https://github.com/NixOS/nixpkgs/pull/156112#issuecomment-1031973946)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yshym** [comment]
Result of `nixpkgs-review pr 160224` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wluma</li>
  </ul>
</details>

**Comment by jmc-figueira** [comment]
The documentation is still pending on https://github.com/NixOS/nixpkgs/pull/156725

Besides that, in https://github.com/NixOS/nixpkgs/pull/156112#issuecomment-1032573487 you mentioned gating the new dependencies based on configuration. Is this not preferable? Am I misunderstanding something?

Besides this, https://github.com/NixOS/nixpkgs/pull/159899 will introduce some changes, but these shouldn't immediately affect this PR.

Otherwise, LGTM.

**Review by jmc-figueira — APPROVED** [review]

**Comment by yshym** [comment]
> The documentation is still pending on #156725
> 
> Besides that, in [#156112 (comment)](https://github.com/NixOS/nixpkgs/pull/156112#issuecomment-1032573487) you mentioned gating the new dependencies based on configuration. Is this not preferable? Am I misunderstanding something?
> 
> Besides this, #159899 will introduce some changes, but these shouldn't immediately affect this PR.
> 
> Otherwise, LGTM.

- #156725 doesn't block this PR. We'll add it after it is merged
- #159899 doesn't block this PR also
- I mentioned [maximbaz's comment](https://github.com/NixOS/nixpkgs/pull/156112#issuecomment-1031973946) in the description of this PR. I thought that we can use a boolean function parameter at first. But the thing is we don't have separate dependencies for loader and driver. We have `vulkan-loader` vulkan dependency only and it is required by the wluma itself, so we can't gate any vulkan dependecies behind a boolean input

**Comment by jmc-figueira** [comment]
> * I mentioned [maximbaz's comment](https://github.com/NixOS/nixpkgs/pull/156112#issuecomment-1031973946) in the description of this PR. I thought that we can use a boolean function parameter at first. But the thing is we don't have separate dependencies for loader and driver. We have `vulkan-loader` vulkan dependency only and it is required by the wluma itself, so we can't gate any vulkan dependecies behind a boolean input

Ah, I see, thanks for the clarification. Upon closer inspection, I realized that what maximbaz was referring to was the fact that, in Arch, there were indeed two dependencies, one of which does not translate to NixOS' use case (vulkan-driver). In that case, there is nothing left do to for this PR as far as I'm aware.

**Comment by jmc-figueira** [comment]
Result of `nixpkgs-review pr 160224` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wluma</li>
  </ul>
</details>

**Comment by yshym** [comment]
> Upon closer inspection, I realized that what maximbaz was referring to was the fact that, in Arch, there were indeed two dependencies, one of which does not translate to NixOS' use case (vulkan-driver)

Correct

**Comment by Lassulus** [comment]
sadly a merge conflict

**Comment by yshym** [comment]
Result of `nixpkgs-review pr 160224` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wluma</li>
  </ul>
</details>

**Pull request merged**
